pkg/api/v1/schema.go

```diff
@@ -174,18 +178,25 @@ type Firmware struct {
 	UUID types.UID `json:"uuid,omitempty"`
 }
 
+// ---
+// +k8s:openapi-gen=true
+type IOThreads struct {
+	// Number of dedicated IOThreads to allocate. Defaults to 1.
+	ThreadCount *uint `json:"threadcount,omitempty"`
```

Comment:
This was a strategy to try and steer the user to use the default ThreadCount. If the user requests using IOThreads but doesn't specify a ThreadCount, we just assume they wanted one OThread. From other discussion that sounds like the most reasonable choice in a majority of cases.

Correct me if I'm wrong, but my impression is that If we made this a uint vs a pointer to one, I think it can no longer be optional and the user would be required to provide a number.

Reply:
Oh, and I made the IOThreads a struct instead of a pointer to an integer directly to act as a sane placeholder for io thread pinning--so that we don't need to mess with the API once we implement that.